p2p/host/basic/basic_host.go

```diff
@@ -143,10 +150,25 @@ func NewHost(net inet.Network, opts *HostOpts) *BasicHost {
 		h.cmgr = opts.ConnManager
 	}
 
+	var relayCtx context.Context
+	var relayCancel func()
+	if opts.Relay {
+		relayCtx, relayCancel = context.WithCancel(context.Background())
+		err := circuit.AddRelayTransport(relayCtx, h, opts.RelayOpts...)
+		if err != nil {
+			// perhaps inappropriate, but otherwise we have to change the interface
+			// to return an error, which will nost likely lead to a fatality anyway
+			panic(err)
```

Comment:
Yeah, relay needs a pre-existing host to attach.
Maybe we can leave it outside the basic host construction if you prefer, although it would be nice to have it simply available through the host options.

Re: panic: I can change the interface to return an error.